src/formatDate.js

```diff
@@ -9,6 +41,31 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const oldDateParts = date.split(oldSeparator);
+
+  let datePartsMatched = {};
+
+  for (let i = 0; i < oldDateParts.length; i++) {
+    datePartsMatched[fromFormat[i]] = oldDateParts[i];
+  }
```

Comment:
The code assumes that the `fromFormat` array will not contain any other elements than 'YYYY', 'YY', 'MM', and 'DD'. Since the task does not specify any restrictions on the possible values within the `fromFormat` array, this could potentially lead to issues if other values are present. It would be more robust to dynamically match the date parts based on the `fromFormat` array without hardcoding the expectations.